src/lambda_function/events/generate_test_event.py

```diff
@@ -0,0 +1,138 @@
+"""
+This is a utility script that generates a fake S3 event notification and writes
+it out as JSON. This test event can be used to test the lambda which
+triggers the Glue pipeline.
+"""
+
+import argparse
+import copy
+import json
+import boto3
+
+SINGLE_RECORD_OUTFILE = 'single-record.json'
+MULTI_RECORD_OUTFILE = 'records.json'
+
+s3_event_record_template = {
+  "eventVersion": "2.0",
+  "eventSource": "aws:s3",
+  "awsRegion": "us-east-1",
+  "eventTime": "1970-01-01T00:00:00.000Z",
+  "eventName": "ObjectCreated:Put",
+  "userIdentity": {
+    "principalId": "EXAMPLE"
+  },
+  "requestParameters": {
+    "sourceIPAddress": "127.0.0.1"
+  },
+  "responseElements": {
+    "x-amz-request-id": "EXAMPLE123456789",
+    "x-amz-id-2": "EXAMPLE123/5678abcdefghijklambdaisawesome/mnopqrstuvwxyzABCDEFGH"
+  },
+  "s3": {
+    "s3SchemaVersion": "1.0",
+    "configurationId": "testConfigRule",
+    "bucket": {
+      "name": "{bucket}",
+      "ownerIdentity": {
+        "principalId": "EXAMPLE"
+      },
+      "arn": "arn:aws:s3:::bucket_arn"
+    },
+    "object": {
+      "key": "{key}",
+      "size": 1024,
+      "eTag": "0123456789abcdef0123456789abcdef",
+      "sequencer": "0A1B2C3D4E5F678901"
+    }
+  }
+}
+
+def read_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+      description="Generate a json file of a mocked SQS event for testing.")
+    parser.add_argument("--input-bucket",
+      default="recover-dev-input-data",
+      help="Optional. S3 bucket name containing input data. Default is 'recover-dev-input-data'.")
+    parser.add_argument("--input-key",
+      default="main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368",
+      help=("Optional. Name of the Synapse dataset containing test data. "
+            "Default is `main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368` "))
+    parser.add_argument("--input-key-prefix",
```

Comment:
I just realize for recover data, if we reprocess, we can only reprocess everything at this point or if we somehow know the specific filename? Are the individual recover zipped files grouped by anything in particular other than the datetime it was created?

Reply:
> Are the individual recover zipped files grouped by anything

That's a discussion we could have with digital health. They are the ones copying the data from the ingestion bucket to the input bucket, so we could come up with a partitioning structure.

In practice we would only ever submit the entire dataset. We need to remove deleted samples and those could appear anywhere.